### PR TITLE
[BUG FIX] Rolling/ExpandingGroupby withMultiIndex Column

### LINKED ISSUE #1032: Groupby with MultiIndex of DataFrame raises ValueError
there is some issue at groupby when MultiIndex column like:

```python
>>> kdf = ks.DataFrame({'a': [1, 2, 3, 2], 'b': [4.0, 2.0, 3.0, 1.0]})
>>> kdf.columns = pd.MultiIndex.from_tuples([('a', 'x'), ('a', 'y')])
>>> kdf
   a
   x    y
0  1  4.0
1  2  2.0
2  3  3.0
3  2  1.0

>>> kdf.groupby(kdf.a.x)
Traceback (most recent call last):
...
ValueError: Cannot combine the series because it comes from a different dataframe. In order to allow this operation, enable 'compute.ops_on_diff_frames' option.
>>> get_option('compute.ops_on_diff_frames')
True
```

it raises error even compute.ops_on_diff_frames option is True.

in pandas, it works well.

```python
>>> pdf = kdf.to_pandas()
>>> pdf.groupby(pdf.a.x).sum()
   a
   x    y
x
1  1  4.0
2  4  3.0
3  3  3.0
```

### PR DESCRIPTION
**Pull request by itholic**
Resolves #1032 

```python
>>> kdf = ks.DataFrame({'a': [1, 2, 3, 2], 'b': [4.0, 2.0, 3.0, 1.0]})
>>> kdf.columns = pd.MultiIndex.from_tuples([('a', 'x'), ('a', 'y')])
>>> pdf = kdf.to_pandas()
>>> kdf.groupby(kdf.a.x).rolling(2).sum().sort_index()
       a
       x    y
x
1 0  NaN  NaN
2 1  NaN  NaN
  3  4.0  3.0
3 2  NaN  NaN
>>>
>>> pdf.groupby(pdf.a.x).rolling(2).sum()
       a
       x    y
x
1 0  NaN  NaN
2 1  NaN  NaN
  3  4.0  3.0
3 2  NaN  NaN
>>>
>>> kdf.groupby(kdf.a.x).expanding(2).sum().sort_index()
       a
       x    y
x
1 0  NaN  NaN
2 1  NaN  NaN
  3  4.0  3.0
3 2  NaN  NaN
>>>
>>> pdf.groupby(pdf.a.x).expanding(2).sum()
       a
       x    y
x
1 0  NaN  NaN
2 1  NaN  NaN
  3  4.0  3.0
3 2  NaN  NaN
```

### CODE REVIEW

File: databricks/koalas/window.py [hunk 1]
```diff
@@ -1226,6 +1234,12 @@ def _apply_as_series_or_frame(self, func):
             _, name = index_map
```
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: @itholic, can you reuse `RollingGroupby._apply_as_series_or_frame` here while we're here? You can do:

```
_apply_as_series_or_frame = RollingGroupby._apply_as_series_or_frame
```

Comment (HyukjinKwon) [reply databricks/koalas/window.py]: For `Expanding` too.

File: databricks/koalas/window.py [hunk 2]
```diff
@@ -586,6 +588,12 @@ def _apply_as_series_or_frame(self, func):
...
             new_index_map.append((SPARK_INDEX_NAME_FORMAT(len(new_index_map)), name))
 
+        if isinstance(kdf.columns, pd.MultiIndex):
```
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: @itholic, all these things should be able to do it only touching metadata in Koalas at internal. Checkout `_InternalFrame(column_index=...)` at `groupby.py`

File: databricks/koalas/window.py [hunk 3]
```diff
@@ -586,6 +588,12 @@ def _apply_as_series_or_frame(self, func):
...
+            groupkey = self._groupkeys[0]
```
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: I think it won't cover the cases such as:

```python
pdf.groupby([pdf.a.x, pdf.a.x]).expanding(2).count()
```

Actually, I think we should have a test for that. If you find some time, please work on it :-).

File: databricks/koalas/window.py (hunk 1 above)
Comment (itholic) [reply databricks/koalas/window.py]: @HyukjinKwon thanks for the review! i applied them :)

File: databricks/koalas/window.py (hunk 3 above)
Comment (itholic) [reply databricks/koalas/window.py]: @HyukjinKwon just fixed logic & and add test cases

File: databricks/koalas/window.py (hunk 2 above)
Comment (itholic) [reply databricks/koalas/window.py]: not pretty sure, i fix the existing logic
could you please check this changing when you available whether it's right or not :)

File: databricks/koalas/tests/test_expanding.py [hunk 4]
```diff
@@ -94,14 +94,17 @@ def _test_groupby_expanding_func(self, f):
...
-        # kdf.columns = pd.MultiIndex.from_tuples([('a', 'x'), ('a', 'y')])
-        # pdf = kdf.to_pandas()
-        # self.assert_eq(
-        #     repr(getattr(kdf.groupby(kdf.a).expanding(2), f)().sort_index()),
-        #     repr(getattr(pdf.groupby(pdf.a).expanding(2), f)()))
+        kdf = ks.DataFrame({'a': [1, 2, 3, 2], 'b': [4.0, 2.0, 3.0, 1.0]})
+        kdf.columns = pd.MultiIndex.from_tuples([('a', 'x'), ('a', 'y')])
+        pdf = kdf.to_pandas()
+        self.assert_eq(
+            repr(getattr(kdf.groupby(kdf.a.x).expanding(2), f)().sort_index()),
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_expanding.py]: I see the problem. `kdf.a` accessor makes another DataFrame and that makes this operations as an operation on different dataframes.

File: databricks/koalas/generic.py [hunk 5]
```diff
@@ -1515,9 +1515,10 @@ def _count_expr(col: spark.Column, spark_type: DataType) -> spark.Column:
 def _resolve_col(kdf, col_like):
     if isinstance(col_like, ks.Series):
         if kdf is not col_like._kdf:
-            raise ValueError(
-                "Cannot combine the series because it comes from a different dataframe. "
-                "In order to allow this operation, enable 'compute.ops_on_diff_frames' option.")
+            if not isinstance(kdf.columns, pd.MultiIndex):
```
Comment (HyukjinKwon) [inline databricks/koalas/generic.py]: In fact, looks we're not currently supporting groupby with different dataframes when `compute.ops_on_diff_frames` is enabled. Seems like we should fix this first.

File: databricks/koalas/tests/test_rolling.py [hunk 6]
```diff
@@ -97,6 +97,18 @@ def _test_groupby_rolling_func(self, f):
...
+        kdf = ks.DataFrame({'a': [1, 2, 3, 2], 'b': [4.0, 2.0, 3.0, 1.0]})
+        kdf.columns = pd.MultiIndex.from_tuples([('a', 'x'), ('a', 'y')])
+        pdf = kdf.to_pandas()
+        self.assert_eq(
+            repr(getattr(kdf.groupby(kdf.a.x).rolling(2), f)().sort_index()),
+            repr(getattr(pdf.groupby(pdf.a.x).rolling(2), f)()))
+
+        self.assert_eq(
+            repr(getattr(kdf.groupby([kdf.a.x, kdf.a.y]).rolling(2), f)().sort_index()),
+            repr(getattr(pdf.groupby([pdf.a.x, pdf.a.y]).rolling(2), f)()))
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_rolling.py]: @itholic, can we just fix the test as below for now:

```python
        # Multiindex column
        kdf = ks.DataFrame({'a': [1, 2, 3, 2], 'b': [4.0, 2.0, 3.0, 1.0]})
        kdf.columns = pd.MultiIndex.from_tuples([('a', 'x'), ('a', 'y')])
        pdf = kdf.to_pandas()
        self.assert_eq(
            repr(getattr(kdf.groupby(("a", "x")).rolling(2), f)().sort_index()),
            repr(getattr(pdf.groupby(("a", "x")).rolling(2), f)()))

        self.assert_eq(
            repr(getattr(kdf.groupby([("a", "x"), ("a", "y")]).rolling(2), f)().sort_index()),
            repr(getattr(pdf.groupby([("a", "x"), ("a", "y")]).rolling(2), f)()))
```

and handle the operations on different DataFrames later in a separate PR.

**Pull request closed without merging**